File: richos/mobile/native-android/core/src/main/kotlin/dev/richos/android/core/AppLinks.kt
package dev.richos.android.core

/**
 * Every address outside the app that RichConnect opens, in ONE place: the privacy policy, support,
 * and this app's Google Play listing (Settings rows, the update notices, `pair-stale`). The names are
 * the iPhone's (`native-ios/Core/Sources/RichOSCore/Settings/AppLinks.swift`), so one list serves both.
 *
 * **PLACEHOLDERS: the CEO fills these before submission.** Neither page is hosted yet (App Store
 * listing drafts, blockers 2 and 3). Each value below is a stand-in; [placeholders] names the ones
 * still unset, so a release check can refuse a build that ships them.
 */
object AppLinks {
    /** PLACEHOLDER(CEO): the hosted privacy policy. Google Play and Apple both require a link inside the app. */
    const val privacyPolicy = "https://richos.ceo/privacy"

    /** PLACEHOLDER(CEO): the support page, with the contact details the stores require. */
    const val support = "https://richos.ceo/support"

    /**
     * This app's Google Play listing, opened in the Play Store app. NOT a placeholder: the
     * application ID `dev.richos.connect` is permanent (the CEO approved it on 2026-09-23).
     */
    fun playStore(applicationId: String) = "market://details?id=$applicationId"

    /** The same listing on the web, for a phone with no Play Store app. */
    fun playStoreWeb(applicationId: String) = "https://play.google.com/store/apps/details?id=$applicationId"

    /** The constants above still holding a stand-in. Empty once the CEO has filled both. */
    val placeholders: List<String> = listOf("privacyPolicy", "support")
}

File: richos/mobile/native-ios/Core/Sources/RichOSCore/Settings/AppLinks.swift
import Foundation

/// Every address outside the app that RichConnect opens, in ONE place: the privacy policy, support,
/// and this app's App Store listing (Settings rows, the update notices, `pair-stale`).
///
/// **PLACEHOLDERS — the CEO fills these before submission.** None of the three final addresses is
/// settled (App Store listing drafts, blockers 1-3: the policy and support pages are not hosted yet,
/// and the App Store Connect record, which assigns the numeric App ID, does not exist). Each value
/// below is a stand-in that opens something harmless; `placeholders` names the ones still unset so a
/// release check can refuse a build that ships them.
public enum AppLinks {
    /// PLACEHOLDER(CEO): the hosted privacy policy (Apple 5.1.1(i) requires a link inside the app).
    public static let privacyPolicy = URL(string: "https://richos.ceo/privacy")!

    /// PLACEHOLDER(CEO): the support page, with the contact details Apple requires.
    public static let support = URL(string: "https://richos.ceo/support")!

    /// PLACEHOLDER(CEO): the numeric Apple ID App Store Connect assigns when the record is created.
    public static let appStoreID = "0000000000"

    /// This app's listing. On iPhone updates come only from the App Store, so "Check for updates"
    /// and "Update in App Store" open it; an apps.apple.com link opens in the App Store app.
    public static var appStore: URL { URL(string: "https://apps.apple.com/app/id\(appStoreID)")! }

    /// The constants above still holding a stand-in. Empty once the CEO has filled all three.
    public static let placeholders: [String] = ["privacyPolicy", "support", "appStoreID"]

    /// Where an "open" effect goes, or `nil` for any other effect.
    public static func destination(of effect: Effect) -> URL? {
        switch effect {
        case .openAppStore: return appStore
        case .openSupport: return support
        case .openPrivacyPolicy: return privacyPolicy
        default: return nil
        }
    }
}
